test/e2e/epp/e2e_test.go

```diff
@@ -204,33 +328,79 @@ func verifyTrafficRouting() {
 	} {
 		ginkgo.By(fmt.Sprintf("Verifying connectivity through the inference extension with %s api and prompt/messages: %v", t.api, t.promptOrMessages))
 
-		// Ensure the expected responses include the InferenceObjective target model names.
-		var expected []string
-		expected = append(expected, targetModelName)
-		curlCmd := getCurlCommand(envoyName, testConfig.NsName, envoyPort, modelName, curlTimeout, t.api, t.promptOrMessages, false)
-
-		actual := make(map[string]int)
+		// Expected ports and InferenceObjective target models
+		expectedPort := generateSequence(firstPort, numPorts)
+		expectedModel := []string{targetModelName}
+
+		// Observed ports and InferenceObjective target models
+		actualModel := make(map[string]int)
+		actualPort := make(map[int]int)
+		// Probability: need to compute estimate of number of batches to send to have high confidence of hitting all ports.
+		// Using the Coupon Collector's Problem formula: n * H_n, where H_n is the nth harmonic number.
+		// This gives us an expected number of trials to collect all coupons (ports).
+		batches := int(math.Ceil(numPorts * harmonicNumber(numPorts)))
+		// Send curl requests to verify routing to all target ports in the InferencePool.
 		gomega.Eventually(func() error {
-			resp, err := testutils.ExecCommandInPod(testConfig, "curl", "curl", curlCmd)
-			if err != nil {
-				return err
+			// Run a small batch per retry (e.g., 5) to keep the test active
+			for i := range batches {
+				uniqueID := time.Now().UnixNano()
+				dynamicHashValue := fmt.Sprintf("Nonce-%d", uniqueID)
+				currentPromptOrMessages := t.promptOrMessages // Start with the original
+
+				// Check if the payload is a slice of maps (e.g., for /chat/completions)
+				if originalMessages, ok := currentPromptOrMessages.([]map[string]any); ok {
+					messagesCopy := make([]map[string]any, len(originalMessages))
+					for idx, msg := range originalMessages {
+						msgCopy := make(map[string]any, len(msg))
+						maps.Copy(msgCopy, msg)
+						// Inject a unique nonce into the content of *EACH* message
+						if content, ok := msgCopy["content"].(string); ok {
+							msgCopy["content"] = fmt.Sprintf("(TestNonce: %s-%d-msg%d) %s", dynamicHashValue, i, idx, content)
+						}
+						messagesCopy[idx] = msgCopy
+					}
+					currentPromptOrMessages = messagesCopy // Use the modified messages for getCurlCommand
+				}
+
+				curlCmd := getCurlCommand(envoyName, testConfig.NsName, envoyPort, modelName, curlTimeout, t.api, currentPromptOrMessages, false)
+
+				resp, err := testutils.ExecCommandInPod(testConfig, "curl", "curl", curlCmd)
```

Comment:
Wrapping `kubectl exec` (which is what` ExecCommandInPod` does) in Go routines adds a lot of complexity (WaitGroups, channels) for a small gain. Since we are only targeting 2 ports, a simple sequential loop is likely enough and much easier to debug.